src/stages/stage5/timeline.c

```diff
@@ -334,197 +135,713 @@ static int stage5_magnetto(Enemy *e, int t) {
-		tsrand_fill(2);
-		create_enemy1c(VIEWPORT_W+200.0*I*afrand(0), 500, Swirl, stage5_miner, -3+2.0*I*afrand(1));
+		PROJECTILE(
+			.proto = pp_plainball,
+			.pos = e->pos,
+			.color = RGBA(0.7, 0.3, 1, 0),
+			.move = move_accelerated(fac * 4 * n, fac * 0.05 * n),
+		);
+		play_sfx_ex("shot_special1", 0, true);
+		WAIT(delay);
 	}
 
-	FROM_TO(1500, 2400, 80)
-		create_enemy1c(VIEWPORT_W*(_i&1)+100.0*I, 300, Fairy, stage5_greeter, 3-6*(_i&1));
+	WAIT(30);
+	e->move = move_linear(-ARGS.acceleration);
+}
 
-	AT(2500) {
-		create_enemy1c(VIEWPORT_W/2, 2000, BigFairy, stage5_lightburst, 2.0*I);
+TASK(lightburst_fairies_1, {
+	int num;
+	cmplx pos1;
+	cmplx pos2;
+	cmplx exit;
+}) {
+	for (int i = 0; i < ARGS.num; i++) {
+		cmplx pos = ARGS.pos1 + ARGS.pos2 * i;
+		INVOKE_TASK(lightburst_fairy_1,
+			.pos = pos,
+			.move_enter = move_towards(pos + ARGS.exit * 70 , 0.05),
+			.move_exit = move_linear(ARGS.exit)
+		);
+		WAIT(40);
 	}
+}
 
-	FROM_TO(2200, 2600, 60-8*global.diff)
-		create_enemy1c(VIEWPORT_W/(6+global.diff)*_i, 200, Swirl, stage5_miner, 3.0*I);
-
-	AT(2700) {
-		if(global.diff > D_Easy) {
-			create_enemy1c(VIEWPORT_W-20+120*I, 2000, BigFairy, stage5_lightburst, -2.0);
-		}
+TASK(lightburst_fairies_2, {
+	int num;
+	cmplx pos1;
+	cmplx pos2;
+	cmplx exit;
+}) {
+	for (int i = 0; i < ARGS.num; i++) {
+		cmplx pos = ARGS.pos1 + ARGS.pos2 * i;
+		INVOKE_TASK(lightburst_fairy_2,
+			.pos = pos,
+			.move_enter = move_towards(pos + ARGS.exit * 70 , 0.05),
+			.move_exit = move_linear(ARGS.exit)
+		);
+		WAIT(40);
 	}
+}
 
-	AT(2900)
-		global.boss = stage5_spawn_midboss();
+TASK(loop_swirl_move, {
+    MoveParams *move;
+	cmplx turn;
+	cmplx retention;
+}) {
+	int begin = creal(ARGS.turn) + 1;
+	int end = cimag(ARGS.turn) - 1;
+	int time = end - begin;
+
+	ARGS.move->retention = ARGS.retention;
+	WAIT(time / 1.5);
+	ARGS.move->retention = 1;
+}
 
-	AT(2920) {
-		stage5_dialog_post_midboss();
+TASK(loop_swirl, {
+	cmplx start;
+	cmplx velocity;
+	cmplx turn;
+	cmplx retention;
+}) {
+	Enemy *e = TASK_BIND(espawn_swirl(ARGS.start, ITEMS(.points = 4, .power = 2)));
 
-		// TODO: determine if this hack is still required with the new dialogue system
-		global.timer++;
+    e->move = move_linear(ARGS.velocity);
+    INVOKE_SUBTASK_DELAYED(80, loop_swirl_move,
+        .move = &e->move,
+		.turn = ARGS.turn,
+		.retention = ARGS.retention
+    );
+
+	int difficulty = 30 - difficulty_value(4, 8, 16, 24);
+
+	for (int x = 0; x < 200; x++) {
+		for(int i = 1; i >= -1; i -= 2) {
+			PROJECTILE(
+				.proto = pp_bullet,
+				.pos = e->pos,
+				.color = RGB(0.3, 0.4, 0.5),
+				.move = move_asymptotic_simple(i * 2 * ARGS.velocity * I / cabs(ARGS.velocity), 3),
+				// TODO: bullets only fire down-left with the above
+				// they should also fire in other directions correlating to the angle of "turn" the swirls make
+				//.rule = asymptotic,
+				//.args = { i*2*e->args[0]*I/cabs(e->args[0]), 3 }
```

Comment:
Sounds like you want to use `e->move.velocity` instead of `ARGS.velocity` then.

Also, you have a `velocity / cabs(velocity)` there, which should be simplified to `cnormalize(velocity)`